querydsl-jpa/src/main/java/com/querydsl/jpa/JPQLTemplates.java

```diff
@@ -99,12 +99,12 @@ protected JPQLTemplates(char escape, QueryHandler queryHandler) {
         add(Ops.MATCHES, "{0} like {1}  escape '" + escape + "'"); // TODO : support real regexes
         add(Ops.MATCHES_IC, "{0} like {1} escape '" + escape + "'"); // TODO : support real regexes
         add(Ops.LOWER, "lower({0})");
-        add(Ops.SUBSTR_1ARG, "substring({0},{1s}+1)", Precedence.ARITH_LOW);
-        add(Ops.SUBSTR_2ARGS, "substring({0},{1s}+1,{2s}-{1s})", Precedence.ARITH_LOW);
+        add(Ops.SUBSTR_1ARG, "substring({0},{1+'1'})", Precedence.ARITH_LOW);
+        add(Ops.SUBSTR_2ARGS, "substring({0},{1+'1'},{2-1})", Precedence.ARITH_LOW);
```

Comment:
The `s` post processing suffix was mostly used to get more stuff for the normalization to convert. Now that the conversion is embedded in the templates it's not anymore necessary.